StoreKitUnitTests/StoreProductTests.swift

```diff
@@ -66,27 +66,51 @@ class StoreProductTests: StoreKitConfigTestCase {
         let productIdentifier = "com.revenuecat.monthly_4.99.1_week_intro"
         let sk1Fetcher = ProductsFetcherSK1(productsRequestFactory: ProductsRequestFactory(),
                                             requestTimeout: Self.requestTimeout)
-        var callbackCalled = false
-
-        sk1Fetcher.products(withIdentifiers: Set([productIdentifier])) { storeProductSet in
-            callbackCalled = true
-            guard let storeProduct = storeProductSet.value?.first else { fatalError("couldn't get product!") }
-
-            expect(storeProduct.productIdentifier) == "com.revenuecat.monthly_4.99.1_week_intro"
-            expect(storeProduct.localizedDescription) == "Monthly subscription with a 1-week free trial"
-            expect(storeProduct.price.description) == "4.99"
-            expect(storeProduct.localizedPriceString) == "$4.99"
-            expect(storeProduct.productIdentifier) == productIdentifier
-            expect(storeProduct.isFamilyShareable) == true
-            expect(storeProduct.localizedTitle) == "Monthly Free Trial"
-            // open the StoreKit Config file as source code to see the expected value
-            expect(storeProduct.subscriptionGroupIdentifier) == "7096FF06"
-
-            expect(storeProduct.subscriptionPeriod?.unit) == .month
-            expect(storeProduct.subscriptionPeriod?.value) == 1
+        var result: Result<Set<StoreProduct>, Error>!
+
+        sk1Fetcher.products(withIdentifiers: Set([productIdentifier])) { products in
+            result = products
         }
 
-        expect(callbackCalled).toEventually(beTrue(), timeout: Self.requestTimeout)
+        expect(result).toEventuallyNot(beNil(), timeout: Self.requestTimeout + .seconds(5))
```

Comment:
👍 
For context for why these used to be separate (i.e.: why have a `callbackCalled` variable), it's mostly force of habit. 
I always do it for methods with nullable results, because that way you can tell whether if it fails whether the problem was a `nil` result or no callback at all. 
But since this method has a non-optional result, there's no value in having `callbackCalled`

Reply:
Makes sense!